snippets/video-render.liquid

```diff
@@ -0,0 +1,82 @@
+{% comment %}
+  Before you start using this snippet file make sure to upload 'components/video-render.js' file in your assets folder!
+
+  Required attribute is 'video' and if 'autoplay' is true - the lazy loading video will be rendered.
+
+  An example of the code to render snippet file:
+
+  {% render 'video-render',
+    video: section.settings.video,
+    poster: url,
+    autoplay: true,
+    controls: false,
+    loop: true,
+    full_width: false,
+    width: 800,
+    height: 600,
+    class: "my-video",
+    part: "video"
+  %}
+
+  Reffering to the 'part' attribute that allows you to style it from any css file:
+
+  // example.css
+  lazy-video::part(video) {}
+
+  .my-class,
+  .my-class::part(video) {}
+{% endcomment %}
+
+{% liquid
+  assign autoplay = autoplay | default: false
+  assign full_width = full_width | default: false
+  assign width = width | default: 1920
+  assign height = height | default: 1080
+  assign poster = poster | default: video.preview_image
+  assign controls = controls | default: false
+  assign muted = muted | default: false
+  assign loop = loop | default: false
+  assign class = class | default: ''
+  assign part = part | default: ''
+%}
+
+{% unless video == null %}
+  {% if autoplay %}
+    <script type="module" src="{{ 'video-render.js' |  asset_url }}" defer="defer"></script>
+
+    {% capture video_sources_value %}
+      [
+        {% for source in video.sources %}
+          {
+            "url": "{{ source.url }}",
+            "mime_type": "{{ source.mime_type }}"
+          }
+        {% endfor %}
+      ]
+    {% endcapture %}
+
+    <lazy-video
+      class="{{ class }}"
+      part="{{ part }}"
+      fullWidth="{{ full_width }}"
+      width="{{ width }}"
+      height="{{ height }}"
+      loop="{{ loop }}"
+      poster="{{ poster |  image_url  }}"
+      sources="{{ video_sources_value |  escape }}"
+    >
+    </lazy-video>
+  {% else %}
+    {{
+      video
```

Comment:
In the preview theme, when I remove the autoplay, the video does not play. It is displayed as a blurry poster video. My thought is that the `video` here is not being passed properly.

Two suggestions:

- [ ] Investigate if the `video` here is being passed correctly
- [ ] To fix the blurry poster, try using width into the param `image_size` ([see here](https://shopify.dev/docs/api/liquid/filters/video_tag#video_tag-image_size))

Screenshot of what I'm seeing.
<img width="1768" alt="Screenshot 2024-08-22 at 9 06 05 AM" src="https://github.com/user-attachments/assets/423f9cec-1f71-40fd-99dd-6810736f6c6a">